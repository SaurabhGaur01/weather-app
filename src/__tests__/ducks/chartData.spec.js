import {
    setChartData,
    clearChartData,
    chartDataReducer,
} from '../../ducks/chartData';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('chartData Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setChartData should match the snapshot' , () => {
            expect(setChartData(mockObject)).toMatchSnapshot();
        });
        it('clearChartData should match the snapshot' , () => {
            expect(clearChartData()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(chartDataReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(chartDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(chartDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value of the API call' , () => {
            expect(chartDataReducer('', setChartData(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearChartData' , () => {
            expect(chartDataReducer(mockObject, clearChartData())).toEqual([]);
        });
    });
});