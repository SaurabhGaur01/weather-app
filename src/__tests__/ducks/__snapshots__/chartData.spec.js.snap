// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chartData Duck Tests Action Generator clearChartData should match the snapshot 1`] = `
Object {
  "type": "modules/weather-app/CLEAR_CHART_DATA",
}
`;

exports[`chartData Duck Tests Action Generator setChartData should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/weather-app/SET_CHART_DATA",
}
`;
